### ISSUE TITLE
Global styles menu: avoid unnecessary mismatch between visible label and accessible name

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#65112]
### Description

See related discussion on https://github.com/WordPress/gutenberg/issues/22332

Cc @joedolson @alexstine 

For the highest level of accessibility, the _visible label_ of a control should determine the control's accessible name.

Sometimes, with good intent, both Core and Gutenberg use the `aria-label` attribute to provide more context but by doing so a mismatch between the visible label and the actual accessible name occurs.

While the expanded info in the `aria-label` may be beneficial for blind screen reader users, it introduces potential barriers for other users. For example:
- Sighted screen reader users will see the visible label but the screen reader will announce the invisible aria-label.
- Speech recognition / Voice control users won't be able to activate the control via a voice command, because of the label / accessible name mismatch, at least with _some_ of the voice control software. See discussion on https://github.com/WordPress/gutenberg/issues/22332

The Web Content Accessibility Guidelines success criterium [Label in Name](https://www.w3.org/TR/WCAG22/#label-in-name) states that the accessible name must _contain_ the visible label and that the best practice is to have the text of the visible label at the start of the name.

Unfortunately, that doesn't work with some voice control software, which makes the WCAG criterium arguably useful.

As such, each time there's a mismatch between visible label and actual accessible name, a new accessibility barrier is introduced for some users. This kind of mismatch should be avoided when possible.

I'd like to reinstate a good general principle:

> [!IMPORTANT]
> When possible, let the visible label of a control be its accessible name. Do not override or expand with screen reader text or aria-label / aria-labelledby.

A good example of _unnecessary_ mismatch is in the Global Styles root menu. Screenshot:

![Screenshot 2024-09-06 at 09 57 46](https://github.com/user-attachments/assets/8c717f66-74eb-40f2-a08c-6690edadf4b5)

Where the buttons _visible labels_ are:

- Typography
- Colors
- Shadows
- Layout

However, these buttons do have [aria-label attributes](https://github.com/WordPress/gutenberg/blob/1a8aa20d057e6bd623d3ff8178b2d67de2845dcc/packages/edit-site/src/components/global-styles/root-menu.js#L44-L64) that only adds the word 'styles'. As such, their accessible name becomes, respectively:

- Typography styles
- Colors styles
- Shadow styles
- Layout styles

This mismatch prevents users of some voice control software to use a voice command.

In this specific case, the added word 'styles' doesn't add much value. These buttons are already placed in a panel that has a heading 'Styles'. The context appears to be clear enough and the mismatch should be avoided. Keeping things simple is always the best option.

When some meaningful, useful, additional context and information are necessary, a better option would be to place the extra info in the control description rather than unnecessarily altering the accessible name.

### Step-by-step reproduction instructions

- Go to Site Editor > Styles
- Observe the root menu in the Styles panel.
- Observe the visible label of the buttons Typography, Colors, Shadows, Layout, mismatches the accessible name provided by the aria-label attributes.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by afercia** [comment]
Same applies to some of the sub-menus, for example the Typography elements:

![Screenshot 2024-09-06 at 10 43 43](https://github.com/user-attachments/assets/59625a98-e3a1-4432-a820-467e09f02ae8)

where, additionally, the `Aa` should be hidden from assistive technology.

**Comment by afercia** [comment]
Some of the sub menus items use a HTML `title` attribute that just repeats the visible text. This is redundant and WordPress has being removing this kind of attributes over the last years, see https://core.trac.wordpress.org/ticket/24766 and all the related tickets and chagesets.

![Screenshot 2024-09-06 at 12 06 41](https://github.com/user-attachments/assets/500a5f5b-40b6-4f73-bab3-a6971452d47a)

**Comment by afercia** [comment]
The Typeset button has an `aria-label={ __( 'Typeset' ) }` that completely mismatches the meaning and visible label:

![Screenshot 2024-09-06 at 14 28 14](https://github.com/user-attachments/assets/9c13413f-5f1c-4764-bd60-cc6111c419e1)

**Comment by afercia** [comment]
The Palette button conditionally shows:
- When there are no palette colors, it shows a visible label 'Add colors'.
- Otherwise, it shows a visual representation of the palette colors. This is not OK for accessibility, as the accessible name provided via an aria-label `Edit palette` is not visually exposed.

![Screenshot 2024-09-06 at 12 10 45](https://github.com/user-attachments/assets/849895c6-4bae-4406-90e3-0bc48eb8e6d1)
